main.js

```diff
@@ -294,6 +319,25 @@ export class IdDecoder {
         bitLength: this.fixedBitLength
       });
     }
+    if(this.multihash) {
+      // <varint hash fn code>: identity function
+      const [hashFnCode] = decoded.slice(0, 1);
+      if(hashFnCode !== MULTIHASH_IDENTITY_FUNCTION_CODE) {
+        throw new Error('Invalid multihash function code.');
+      }
+      // <varint digest size in bytes>: 32
+      const [digestSize] = decoded.slice(1, 2);
+
+      if(digestSize !== KEY_SEED_BYTE_SIZE) {
```

Comment:
This doesn't seem right -- as a different size could be used. I recommend we pass in an optional `options` param to `decode` that sets the default "expected size" to 32 bytes, but allows for a different expected size to be passed as well.

Reply:
Updated here abe8865